File: HW/HW 2/Solution.java
public class Solution{
    public boolean isBadVersion(int n){ // created my own method for testing purposes based on the leet code problem
        return n>=4;
    }
    public int firstBadVersion(int n) {
        int left = 1;
        int right = n;
        while(left<right){
            int mid = left+(right-left)/2;
            if(isBadVersion(mid)){
                right = mid;
            }
            else{
                left = mid+1;
            }
        }
        return left;
    }

    public static void main(String Args[]){
        Solution solution = new Solution();
        int n = 10;
        System.out.println(solution.firstBadVersion(n));
    }
}

File: HW/HW 2/Solution.py
class Solution:
    def __init__(self):
        pass

    def isBadVersion(self, version):
        # Assuming version 4 is the first bad version
        return version >= 4

    def firstBadVersion(self, n):
        left, right = 1, n
        while left < right:
            mid = left + (right - left) // 2
            if self.isBadVersion(mid):
                right = mid
            else:
                left = mid + 1
        return left

if __name__ == "__main__":
    solution = Solution()
    n = 10
    print(solution.firstBadVersion(n))
